src/jit/flowgraph.cpp

```diff
@@ -4309,6 +4309,7 @@ void Compiler::fgSwitchToOptimized()
     assert(fgCanSwitchToOptimized());
 
     // Switch to optimized and re-init options
+    JITDUMP("****\n**** JIT Tier0 jit request switching to Tier1 because of loop\n****\n");
```

Comment:
`request switching to Tier1 ` It is not TIer1 actually.
We should find a name for a mode when both `JitFlags::JIT_FLAG_TIER0` and `JitFlags::JIT_FLAG_TIER1` are unset, I called it `No-Tier` in https://github.com/dotnet/coreclr/pull/27017.

Reply:
It is likely just normal full optimization, which is currently the same as Tier1.

Reply:
> It is likely just normal full optimization, which is currently the same as Tier1.

We have at least 2 statements that are currently true:
1. that we don't have `JitFlags::JIT_FLAG_TIER0`  and `JitFlags::JIT_FLAG_TIER1` both set at the same time;
2. statement that  `JitFlags::JIT_FLAG_TIER1` generate the same code as No-Tier (`JitFlags::JIT_FLAG_TIER0` and `JitFlags::JIT_FLAG_TIER1` both unset);

Could we confirm the first statement with asserts and change the second to `JitFlags::JIT_FLAG_TIER1` is always set when `JitFlags::JIT_FLAG_TIER0` is not (even in crossgen)?

Reply:
I think we should simply describe this for what it is -- a Tier0 request that the jit decided to convert to full optimization because of a loop.

Someday down the road the codegen full opts and Tier1 may diverge; in practice even today they likely diverge in practice since Tier1 requests happen after a lot of runtime state has been set up, and that state can alter how the jit generates code.

Reply:
> statement that JitFlags::JIT_FLAG_TIER1 generate the same code as No-Tier 

hm.. I was told recently it's not true: https://github.com/dotnet/coreclr/pull/27023#issuecomment-538198467

Reply:
@egorbo Today the jit has the same optimizations enabled for full opts and Tier1. But the jit consults the runtime as it makes optimization decisions, and the runtime state will differ depending on when the method is jitted.

Since Tier1 methods are jitted later they often end up with better codegen, since more classes are initialized, readonly static values are known, etc.

Reply:
@sandreenko I propose we merge this and sort out the tier0/loop thing subsequently....